tensorflow/lite/experimental/micro/examples/mnist_demo/main.cc

```diff
@@ -0,0 +1,140 @@
+void printTensorDetails(TfLiteTensor* tensor,
+                        tflite::ErrorReporter* error_reporter) {
+  error_reporter->Report("Type [%s] Shape :", typeString(tensor->type));
+  for (int d = 0; d < tensor->dims->size; ++d) {
+    error_reporter->Report("%d [ %d]", d, tensor->dims->data[d]);
+  }
+  error_reporter->Report("");
+}
+
+int main(int argc, char* argv[]) {
+  // Set up logging.
+  tflite::MicroErrorReporter micro_error_reporter;
+  tflite::ErrorReporter* error_reporter = &micro_error_reporter;
+
+  // Map the model into a usable data structure. This doesn't involve any
+  // copying or parsing, it's a very lightweight operation.
+  error_reporter->Report("Parsing MNIST classifier model FlatBuffer.");
+  const tflite::Model* model = ::tflite::GetModel(mnist_dense_model_tflite);
+  if (model->version() != TFLITE_SCHEMA_VERSION) {
+    error_reporter->Report(
+        "Model provided is schema version %d not equal "
+        "to supported version %d.\n",
+        model->version(), TFLITE_SCHEMA_VERSION);
+    return 1;
+  }
+  error_reporter->Report("Model parsed.\n");
+
+  // This pulls in all the operation implementations we need.
+  tflite::ops::micro::AllOpsResolver resolver;
+
+  // Create an area of memory to use for input, output, and intermediate arrays.
+  // The size of this will depend on the model you're using, currently
+  // determined by experimentation.
+  const int tensor_arena_size = 200 * 1024;
+  uint8_t tensor_arena[tensor_arena_size];
+
+  // Build an interpreter to run the model with.
+  error_reporter->Report("Build interpreter.");
+  tflite::MicroInterpreter interpreter(model, resolver, tensor_arena,
+                                       tensor_arena_size, error_reporter);
+  interpreter.AllocateTensors();
+  error_reporter->Report("Done.\n");
+
+  // Get information about the models input and output tensors.
+  TfLiteTensor* model_input = interpreter.input(0);
+  error_reporter->Report("Details of input tensor:");
+  printTensorDetails(model_input, error_reporter);
+  TfLiteTensor* model_output = interpreter.output(0);
+  error_reporter->Report("Details of output tensor:");
+  printTensorDetails(model_output, error_reporter);
+
+  // perform inference on each test sample and evalute accuracy of model
+  int accurateCount = 0;
+  for (int s = 0; s < mnistSampleCount; ++s) {
+    // Set value of input tensor
+    for (int d = 0; d < 784; ++d) model_input->data.f[d] = mnistInput[s][d];
+
+    // perform inference
+    error_reporter->Report("Performing inference on sample %d", s);
+    TfLiteStatus invoke_status = interpreter.Invoke();
+    if (invoke_status != kTfLiteOk) {
+      error_reporter->Report("Invoke failed.\n");
+      return 1;
+    }
+
+    error_reporter->Report("Model output [%d] training label [%d]",
+                           model_output->data.i32[0], mnistOutput[s]);
+
+    if (model_output->data.i32[0] == mnistOutput[s]) ++accurateCount;
```

Comment:
Same as above with for loops - our linter requires:
if (...) {
...
}

Reply:
Updated coding style.